src/tmpfiles/tmpfiles.c

```diff
@@ -1525,13 +1525,16 @@ static const char *creation_mode_verb_table[_CREATION_MODE_MAX] = {
 
 DEFINE_PRIVATE_STRING_TABLE_LOOKUP_TO_STRING(creation_mode_verb, CreationMode);
 
-static int create_directory_or_subvolume(const char *path, mode_t mode, bool subvol) {
+static int create_directory_or_subvolume(const char *path, mode_t mode, bool subvol, CreationMode *creation) {
```

Comment:
hmm, I think this would be simpler if we'd just use the return value for this. return == 0 when the directory/subvol already existed, and all is good but we did nothing, or > 0 if the directory/subvol was created and we actually did something. A similar return protocol we use in quite a number of cases and I think it would be appropriate here too.

Reply:
ah, hmm, maybe not, this returns an fd after all. ignore the above then